lib/routes/charge/items/item_page.dart

```diff
@@ -14,29 +14,53 @@ import 'package:breez/widgets/single_button_bottom_bar.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 
-class CreateItemPage extends StatefulWidget {
+import '../currency_wrapper.dart';
+
+class ItemPage extends StatefulWidget {
+  final Item item;
+  final AccountModel accountModel;
   final PosCatalogBloc _posCatalogBloc;
 
-  CreateItemPage(this._posCatalogBloc);
+  ItemPage(this._posCatalogBloc, {this.item, this.accountModel});
 
   @override
   State<StatefulWidget> createState() {
-    return CreateItemPageState();
+    return ItemPageState();
   }
 }
 
-class CreateItemPageState extends State<CreateItemPage> {
+class ItemPageState extends State<ItemPage> {
   final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
 
   final _formKey = GlobalKey<FormState>();
   TextEditingController _nameController = TextEditingController();
   TextEditingController _priceController = TextEditingController();
+  TextEditingController _skuController = TextEditingController();
   AccountBloc _accountBloc;
   bool _isInit = false;
-  bool _isFiat = false;
-  Currency _selectedCurrency = Currency.BTC;
-  FiatConversion _selectedFiatCurrency;
-  String currency;
+  bool _inEditMode = false;
+  String _titlePrefix = "Add";
+  CurrencyWrapper _selectedCurrency = CurrencyWrapper.fromBTC(Currency.BTC);
+
+  @override
+  void initState() {
+    super.initState();
+    _inEditMode = widget.item != null;
+    if (_inEditMode) _initializeFields();
```

Comment:
Maybe initialize the fields anyway? In the "new" mode we need to change the title anyway (I think it is better to have  a variable "title" instead of "titlePrefix").
We can embed here the initialization without needing another function.

Reply:
Title changes:
3d59c89
5a7e6f5